replaced screenshots using dark theme, added "Features" section (#295) (#297)
redhat-developer/intellij-kubernetes: README.md

## Overview

JetBrains IDEA plugin that allows to browse, create, modify and delete resources on Kubernetes or OpenShift clusters.
The plugin provides functionalities and user experiences that are very close to the Kubernetes extension for vscode (https://marketplace.visualstudio.com/items?itemName=ms-kubernetes-tools.vscode-kubernetes-tools).

It is available for install from the [JetBrains Marketplace](https://plugins.jetbrains.com/plugin/15921-kubernetes-by-red-hat).

![editor push scenario](images/editor-push.gif)

> **NOTE:** This plugin is in Preview mode. The plugin support for Kubernetes or OpenShift clusters is strictly experimental - assumptions may break, commands and behavior may change!

> ## Key Features

Author: @mohitsuman much better than my text, indeed.

* Browse Kubernetes and OpenShift cluster resources in dynamically update tree
* Delete resources on cluster
* Edit resources on cluster backed by kubernetes resource schema
* Create new resources on cluster
* Editor kept in sync with the resource on cluster by pushing & pulling

## Features

### Browse Kubernetes & OpenShift resources
The plugin offers a tool window with a tree that displays all the resources that exist on a Kubernetes or OpenShift cluster. 
The tree is updated as the resources on the cluster change so that it always display the current state of the resources. 

#### Current Context
The tree displays resources that exist on the cluster that the current context points to. 
The user can switch the current context by choosing any context that exists and picking "Set as Current Cluster" in the context menu.
OpenShift clusters are shown as such with an OpenShift icon instead of a Kubernetes one.
 
#### Current Namespace/Project
The tree displays resources that exist within the current namespace/project as specified in the kube config. 

> Of course resources that live outside the scope of namespace/project are displayed, too.

Author: it is talking of cluster scoped resources aka non-namespaced resources.

> Of course resources that live outside the scope of namespace/project are displayed, too.

Reviewer: I would frame it as `Non-namespaced aka cluster-scoped resources are also listed in the tree view`

> You can switch the current namespace/project with the context menu item "Use Namespace". 

Author: Not sure it is better to talk of the "Users" than to talk to the users ("You").

In OpenShift clusters the tree also lists projects where the tooling behaves in an equivalent manner. 

#### Resource Categories
Resources are grouped in different categories such as Namespaces, Nodes, Workloads, Network, Storage, Configuration and Custom Resources.

![img.png](images/categories.png)

#### Pods
Pods are shown with their running state. When running their icon holds a green dot. 
When red, the dot indicates that a pod is either pending, succeeded (terminated), failed or is in an unknown state.
Pods also unveil their IP address and the number of running containers out of the total number of containers.

![img.png](images/pod.png)

### Create and Edit Resources

> You can edit any existing resource that is shown in the resource tree.

Author: @mohitsuman what do you mean by
>resource manifests
?
We're editing the resources, aren't we?

> You can edit any existing resource that is shown in the resource tree.

Reviewer: Yes let's keep it `resource` only.

```suggestion
You can edit Kubernetes resources that is shown in the resource tree and apply them to the cluster.
```

> You can edit any existing resource that is shown in the resource tree.

Author: `is shown` -> `are shown`

Either double click it or pick "Edit..." in the context menu. 
A Yaml editor opens up and allows you to change the resource. 
Pushing the editor to the cluster then updates the resource that you were editing.

> You can also open a local yaml/json file and create a new resource on the cluster by pushing the editor to it.

Author: Here again you talk of a 3rd person, the "user" instead of talking to the user. I'm not convinced that this is better.

> **Tip**: You're advised to edit copies of your yaml/json files. 
> This is because the editor keeps your content in sync with the resource that exists on the cluster. 
> Changes that happen on the cluster are included when pushing or pulling the editor.  
    

![editor](images/editor.png)

The editor validates your content on behalf of a schema that's matching your kubernetes resource.
For instance if you're editing a pod, the editor validates it by a kubernetes pod schema and highlights the portions that are infringing.

![editor schema](images/editor-schema.png)

#### Push (Save) to Cluster

You can push your editor to the cluster in order to update an existing resource or create a new one.

> The cluster that the editor is saved to is set in the current context (cluster, namespace and user) that's defined in your kube config.

Reviewer: ```suggestion
Once the changes are deployed on the cluster, the extension sets the current context (cluster, namespace and user) pointing to the cluster as defined in the kubeconfig.
```

Author: I guess that my initial text was misleading: I read in your text that the editor is setting the current context which isn't true. I was trying to express that the editor was saving to the current context that's set in the kubeconfig.

Reviewer: Thanks for clarifying this @adietish. I think the new wording works. 👍🏼

Reviewer: ```suggestion
The editor is saving the changes to the current context (cluster, namespace and user) that's defined in the kubeconfig
```

> The editor has a toolbar with an action for this sake.

Reviewer: ```suggestion
There is a `Push` button available in the editor view to make the action more discoverable.
```

It is available for install from the [JetBrains Marketplace](https://plugins.jetbrains.com/plugin/15921-kubernetes-by-red-hat).

![editor push scenario](images/editor-push.gif)

> **NOTE:** This plugin is in Preview mode. The plugin support for Kubernetes or OpenShift clusters is strictly experimental - assumptions may break, commands and behavior may change!

## Key Features
 - Edit Kubernetes resource manifests and apply them to your cluster
 - View Kubernetes and OpenShift clusters in an explorer tree view
 - Tree View supports dynamic update of K8s resources
 - Allow Push and Pull Feature to keep the editor in sync with the Kubernetes resource
 - Support Creation and Deletion of new Kubernetes resources on Cluster
 - Navigate into Kubernetes workloads, services, pods and nodes
 - Switching contexts and namespaces within Kubernetes
* Browse Kubernetes and OpenShift cluster resources in dynamically update tree
* Delete resources on cluster
* Edit resources on cluster backed by kubernetes resource schema
* Create new resources on cluster
* Editor kept in sync with the resource on cluster by pushing & pulling

> 

Reviewer: @adietish There are some duplicate lines being added. We need to remove the ones starting with '*'.

Author: @mohitsuman thx for spottig. Looks like merging via github has problems. Correcting those problem by lookup up your comments above.

Kubernetes Resources that exist within the current Namespace/Project (as specified in kubeconfig file) are displayed in the tree view 
Of course resources that live outside the scope of namespace/project are displayed, too.
Users can switch the current Namespace/Project using the context menu action `Use Namespace`
In OpenShift clusters the tree also lists projects where the tooling behaves in an equivalent manner. 

#### Resource Categories
Resources are grouped in different categories such as Namespaces, Nodes, Workloads, Network, Storage, Configuration and Custom Resources.

![img.png](images/categories.png)

#### Pods
Pods are shown with their running state. When running their icon holds a green dot. 
When red, the dot indicates that a pod is either pending, succeeded (terminated), failed or is in an unknown state.
Pods also unveil their IP address and the number of running containers out of the total number of containers.
The pod in the container is identified with color based on the state of the pod.
- A green dot indicates that the pod is currently running.
- A red dot indicates that the pod is in Pending, Succeeded (Terminated), Failed or in Unknown State.

> ![img.png](images/pod.png)

Reviewer: @adietish Here also the lines are repeated. We need to remove line no 51,52. And move 54-56 above line 53.

Author: @mohitsuman is this what your instructions result into? Please confirm:
```
The pod in the container is identified with color based on the state of the pod.
- A green dot indicates that the pod is currently running.
- A red dot indicates that the pod is in Pending, Succeeded (Terminated), Failed or in Unknown State.  

![img.png](images/pod.png)

Pods also unveil their IP address and the number of running containers out of the total number of containers.
```

Reviewer: Yes this is the intended change.
